tools/makemanifest.py: Provide access to MICROPY_MANIFEST_xxx symbols.

robert-hh opened the pull request:
The new function resolve(name) returns the value of a Makefile symbol defined as MICROPY_MANIFEST_<name>.

Example:

Makefile:

MICROPY_MANIFEST_FROZEN_SET = "default"

manifest.py:

frozen_set = resolve("$FROZEN_SET")

If name is not defined, resolve() returns `None`.

Notes:
- Instead of "resolve" the function may have any other appropriate name.
- The "$XYZ" notation was kept for consistency with the other functions.

P.S.: In case that this feature existed already, just give me a hint and sorry for the wasting your time.

robert-hh committed:
tools/makemanifest.py: Provide access to MICROPY_MANIFEST_xxx symbols.

The new function resolve(name) returns the value of a Makefile
symbol defined as MICROPY_MANIFEST_<name>.

Example:

Makefile:

MICROPY_MANIFEST_FROZEN_SET = "default"

manifest.py:

frozen_set = resolve("$FROZEN_SET")

If name is not defined, resolve() returns `None`. Set the symbols
MICROPY_MANIFEST_BOARD and MICROPY_MANIFEST_BOARD_VARIANT by
default.

Notes:
- Instead of "resolve" the function may have any other appropriate
  name.
- The "$XYZ" notation was kept for consistency with the other functions.

Signed-off-by: robert-hh <robert@hammelrath.com>

jimmo commented:
@robert-hh Thanks for this!

This was basically what I was considering doing (although I was also thinking of just putting these variables directly into `globals`), but the issue I ran into is that it's currently assumed (by `makemanifest.py`) that all variables are paths, and it does:

```
    if os.path.exists(value):
        value = os.path.abspath(value)
```

I'm not sure this is the right behavior, although it kind of "works by accident" with the `BOARD` and `BOARD_VARIANT` variables added in this PR because they will fail `os.path.exists`.

I can see two obvious options:
* Only do the `abspath` when the variable actually gets used as a path (i.e. in the `_resolve_path` function in `manifestfile.py`)
* Have a different convention for path vs non-path variables (i.e. `MICROPY_MANIFEST_PATH_FOO` vs `MICROPY_MANIFEST_VAR_FOO`).

robert-hh committed:
tools/makemanifest.py: Expand the path only for known directories.

That imposes the least side effects and does not add a naming
convention. If a manifest.py script want to expand a path, it can do
so using e.g. os.path.abspath().

Signed-off-by: robert-hh <robert@hammelrath.com>

robert-hh commented:
Closing due to minor importance and to lower the PR count.
